src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,158 @@
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                return Optional.of(extractOrderFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get order from DB with ID = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * FROM orders;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                orders.add(extractOrderFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get orders from DB", e);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        String queryToDeleteProducts = "DELETE FROM orders_products WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToDeleteProducts)) {
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete product from shopping cart - "
+                    + order, e);
+        }
+        return insertProducts(order);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders SET deleted = true WHERE order_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete order with ID = " + id, e);
+        }
+    }
+
+    private Order insertProducts(Order order) {
+        String queryToUpdateProducts = "INSERT INTO orders_products(order_id, product_id) "
+                + "VALUES(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToUpdateProducts)) {
+            for (int i = 0; i < order.getProducts().size(); i++) {
+                statement.setLong(1, order.getId());
```

Comment:
We don't need to do this insert inside loop - value is the same for all products

Reply:
Agree to the first part. But somehow foreach didn't work, the table was empty. Still, it works well with this loop